### ISSUE TITLE
Bean reference fields should have an addition button next to it allowing to jump to the entry in the beans tab

### ISSUE THREAD

**Issue by lhein** [KaotoIO/kaoto-next#744]
Currently Bean Ref fields in the config form are displayed as drop down with the possibility to select an existing bean or creating a new one. Can we show a button on the right of the drop down which allows the following:

- if no bean is currently selected, it will just jump to the bean tab
- if a bean is currently selected, it will jump to the bean tab and select the corresponding  bean there

I don't know how hard it is but it would be convenient if we offer kind of a "go back" link / button, which allows you to go back to the design tab and be on the same screen you left before.

**Comment by igarashitm** [comment]
w/r/t going back to the config form, we can show full beans editor in the modal. I don't know if it's feasible to fully route to the Beans page/tab and put a link to come back to the Design page/tab in exactly same state. @lordrip do you have an idea if it's feasible and if it's worth doing it but not in a modal?

Also see: 
 - https://github.com/KaotoIO/kaoto-next/issues/558#issuecomment-1862804706

**Comment by lhein** [comment]
That could be also an option, Tomo...using the modal.

**Comment by lordrip** [comment]
> w/r/t going back to the config form, we can show full beans editor in the modal. I don't know if it's feasible to fully route to the Beans page/tab and put a link to come back to the Design page/tab in exactly same state. @lordrip do you have an idea if it's feasible and if it's worth doing it but not in a modal?
> 
> Also see:
> 
>     * [Renaming the bean in source code not reflected in the form and bean reference in route source  #558 (comment)](https://github.com/KaotoIO/kaoto-next/issues/558#issuecomment-1862804706)

Yes, it is possible, we just need to rely more on the `Router`. Now, for this to work in both scenarios, we would need to use the Router also in the VSCode wrapper as well.
